Source/JavaScriptCore/runtime/StringPrototype.cpp

```diff
@@ -719,10 +719,11 @@ JSC_DEFINE_HOST_FUNCTION(stringProtoFuncSplitFast, (JSGlobalObject* globalObject
             auto view = thisString->view(globalObject);
             RETURN_IF_EXCEPTION(scope, { });
 
+            bool encounteredNonAtoms = false;
             for (unsigned i = 0; i < resultSize; ++i) {
                 unsigned end = result[i];
                 JSString* string = nullptr;
-                if (makeAtomStringsArray) {
+                if (makeAtomStringsArray && !(start < end && isASCIIDigit(view->characterAt(start)))) {
```

Comment:
It would be more effective to check if it’s not an identifier start, which for ascii is $A-Za-z_. So anything not in that range which also excludes characters like ^ and importantly quote characters like `”` `’` and commas